src/test/java/net/imglib2/algorithm/hough/HoughLineTransformTest.java

```diff
@@ -0,0 +1,185 @@
+ * {@link HoughTransforms#getSlope}, and {@link HoughTransforms#getIntercept}.
+ *
+ * @author Yili Zhao
+ * @author Gabe Selzer
+ */
+public class HoughLineTransformTest
+{
+
+	private BufferedImage groundTruth;
+
+	private Img< UnsignedByteType > tpImg;
+
+	public static int N_THREADS = Runtime.getRuntime().availableProcessors();
+
+	@Before
+	public void setUp() throws IOException
+	{
+		groundTruth = ImageIO.read( getClass().getResource( "result9.png" ) );
+		tpImg = ArrayImgs.unsignedBytes( 11l, 13l );
+		final RandomAccess< UnsignedByteType > randomAccess = tpImg.randomAccess();
+		// create vertical line
+		randomAccess.setPosition( 5, 0 );
+		for ( int i = 0; i < 13; i++ )
+		{
+			randomAccess.setPosition( i, 1 );
+			randomAccess.get().set( 255 );
+		}
+
+		// create horizontal line
+		randomAccess.setPosition( 5, 1 );
+		for ( int i = 0; i < 11; i++ )
+		{
+			randomAccess.setPosition( i, 0 );
+			randomAccess.get().set( 255 );
+		}
+
+		// create diagonal line
+		for ( int i = 0; i < 11; i++ )
+		{
+			randomAccess.setPosition( i, 0 );
+			randomAccess.setPosition( i / 3, 1 );
+			randomAccess.get().set( 255 );
+		}
+	}
+
+	@Test
+	public < T extends RealType< T > > void testHoughLineTransformToTarget() throws IOException
+	{
+		// create votespace
+		final long[] dims = new long[ tpImg.numDimensions() ];
+		tpImg.dimensions( dims );
+		final long[] outputDims = HoughTransforms.getVotespaceSize( tpImg );
+		final Img< UnsignedByteType > votespace = new ArrayImgFactory().create( outputDims, tpImg.firstElement() );
+
+		// run transform
+		HoughTransforms.voteLines( tpImg, votespace );
+
+		// compare expected / actual dimensions
+		int height = groundTruth.getHeight();
+		int width = groundTruth.getWidth();
+		assertEquals( "Ground truth and result height do not match.", height, outputDims[ 1 ] );
+		assertEquals( "Ground truth and result width do not match.", width, outputDims[ 0 ] );
+
+		DataBufferByte db = ( DataBufferByte ) groundTruth.getData().getDataBuffer();
+		byte[] expected = db.getBankData()[ 0 ];
+
+		final RandomAccess< UnsignedByteType > ra = votespace.randomAccess();
+
+		// compare expected / actual results for regression.
+		int index = 0;
+		for ( int j = 0; j < height; ++j )
+		{
+			ra.setPosition( j, 1 );
+			for ( int i = 0; i < width; ++i )
+			{
+				ra.setPosition( i, 0 );
+
+				assertEquals( expected[ index++ ], ra.get().getRealDouble(), 0 );
+
+			}
+		}
+	}
+
+	@Test
+	public < T extends RealType< T > > void testPickPeaks()
+	{
+		// three expected peaks
+		final long[] expected = { -5, -89, -5, -88, 0, -76, 0, -74, 0, -73, 5, -2, 5, -1, 5, 0, 5, 1, 5, 2, 5, 88 };
+		int index = 0;
+
+		// create votespace
+		final long[] dims = new long[ tpImg.numDimensions() ];
+		tpImg.dimensions( dims );
+		final long[] outputDims = HoughTransforms.getVotespaceSize( tpImg );
+		final Img< UnsignedByteType > votespace = new ArrayImgFactory().create( outputDims, tpImg.firstElement() );
```

Comment:
Use `ArrayImgs.unsignedBytes( outputDims )` instead of `new ArrayImgFactory().create`